src/wallet.js

```diff
@@ -1235,14 +1274,16 @@ const wallet = {
    * @param {Object} historyTransactions Object of transactions indexed by tx_id
    * @param {Object} Array with all tokens already selected in the send tokens
    *
-   * @return {Object} {success: boolean, message: error message in case of failure, data: prepared data in case of success}
+   * @return {Promise} Promise resolving an object {success: boolean, message: error message in case of failure, data: prepared data in case of success}
    *
    * @memberof Wallet
    * @inner
    */
-  prepareSendTokensData(data, token, chooseInputs, historyTransactions, allTokens) {
+  async prepareSendTokensData(data, token, chooseInputs, historyTransactions, allTokens) {
     // Get the data and verify if we need to select the inputs or add a change output
-
+    // Need to use a async/await because we get the current height of the blockchain
+    // from api so we can check if a block reward can be spent
+    const response = await walletApi.getMiningInfoRequest();
```

Comment:
You're calling it in multiple places. Maybe you should schedule an update every N seconds, and create a `getMiningInfoRequestIfNeeded()` that will only call if at most X seconds has passed. What do you think?

Reply:
I just call it if is not passed in the parameter. In some methods this is useful because we call `canUseUnspentTx` many times, so I execute `getMiningInfoRequest` only once and pass as parameter.

See method `prepareSendTokensData`. I call `canUseUnspentTx` in a loop, so instead of requesting the data from the API every call I execute one time in `prepareSendTokensData` and pass as parameter.

Reply:
I think using a loop won't be needed as we get the information on websocket regularly, so we'll probably always call the methods with the height parameter. But the code to get this info from websocket is on the wallet PR, right?

Reply:
Humm your idea would be to get the ws message from the full node in the lib and update internally so we wouldn't need to get this data from the API anymore?

This could work and the code would be much easier. When I opened this PR we didn't have this ws message but now that we have it might be better.

Reply:
The idea was to use this info which is already available on the wallet. But "intercepting" it directly on the lib seems like a better way to handle it.